lib/buffer.js

```diff
@@ -107,7 +111,7 @@ function Buffer(arg, encodingOrOffset, length) {
         'If encoding is specified then the first argument must be a string'
       );
     }
-    return Buffer.allocUnsafe(arg);
+    return allocate(arg);
```

Comment:
@RReverser 
> your reasoning above seemed to be that it wouldn't throw on negative numbers if I keep as-is

Yes, but only because there’s the future plan (#7079) of making `Buffer.allocUnsafe()` throw for negative input; if you’re referring to the current situation, you’re right, there’s no difference here.